Sources/LeafKit/LeafError.swift

```diff
@@ -1,3 +1,6 @@
+// MARK: Subject to change prior to 1.0.0 release
+// MARK: -
```

Comment:
* Annotation/Rearrangment
* Makes `LexerError.lexed` internal to allow `LeafToken` to be internal
* Renames `LexerError.internalError` to `LexerError.unknownError`